hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -336,13 +334,13 @@ private TransactionBody handleUserTransaction(
                 recordBuilder.status(preCheckResult.responseCodeEnum());
 
             } else {
+                networkUtilizationManager.trackTxn(transactionInfo, consensusNow, state);
                 feeAccumulator.charge(payer, fees);
                 try {
-                    if (!isPayerThrottleExempt && networkUtilizationManager.wasLastTxnGasThrottled()) {
-                        // Refund the service fees already charged to the payer, because the user-submitted transaction
-                        // was fully valid but network capacity was unavailable to satisfy it.
-                        final var serviceFee = new Fees(0L, 0L, fees.serviceFee());
-                        feeAccumulator.refund(payer, serviceFee);
+                    if (networkUtilizationManager.wasLastTxnGasThrottled()) {
+                        // Don't charge the payer the service fee component, because the user-submitted transaction
+                        // was fully valid but network capacity was unavailable to satisfy it
+                        fees = new Fees(fees.nodeFee(), fees.networkFee(), 0L);
```

Comment:
Should we also set the `payer` here? The idea is that currently it might work as expected, but someone can write a code that sets `payer` to something else in the future - we don't have control over it. If we set the `fees` and `payer` and then throw the exception we are sure that we are going to charge the correct user the correct amount.